### DISCUSSION TITLE
Default synchronize_session?

### DISCUSSION THREAD

**Discussion by andrew222651** [sqlalchemy/sqlalchemy#8637] [Usage Questions] [answered]
The [docs](https://docs.sqlalchemy.org/en/14/orm/session_basics.html#selecting-a-synchronization-strategy) say that `synchronize_session` can be set on update and delete statements. Is there a way of setting it for an entire session?

**Comment by zzzeek** [comment, marked as answer]
not yet (at least, not without using an event hook which is slightly more verbose).   #7837 will accommodate this.

**Reply by zzzeek** [reply]
also, in SQLAlchemy 2.0 which is now released as beta1, synchronize_session defaults to "auto" which will probably do the best thing in each case.
